To: vim_dev@googlegroups.com
Subject: Patch 7.3.327
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.327
Problem:    When jumping to a help tag a closed fold doesn't open.
Solution:   Save and restore KeyTyped. (Yasuhiro Matsumoto)
Files:	    src/ex_cmds.c


*** ../vim-7.3.326/src/ex_cmds.c	2011-09-14 14:43:21.000000000 +0200
--- src/ex_cmds.c	2011-09-30 17:24:03.000000000 +0200
***************
*** 5508,5513 ****
--- 5508,5516 ----
      int		len;
      char_u	*lang;
  #endif
+ #ifdef FEAT_FOLDING
+     int		old_KeyTyped = KeyTyped;
+ #endif
  
      if (eap != NULL)
      {
***************
*** 5671,5676 ****
--- 5674,5685 ----
      if (!p_im)
  	restart_edit = 0;	    /* don't want insert mode in help file */
  
+ #ifdef FEAT_FOLDING
+     /* Restore KeyTyped, setting 'filetype=help' may reset it.
+      * It is needed for do_tag top open folds under the cursor. */
+     KeyTyped = old_KeyTyped;
+ #endif
+ 
      if (tag != NULL)
  	do_tag(tag, DT_HELP, 1, FALSE, TRUE);
  
*** ../vim-7.3.326/src/version.c	2011-09-30 16:56:00.000000000 +0200
--- src/version.c	2011-09-30 17:24:17.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     327,
  /**/

-- 
Q. What happens to programmers when they die?
A: MS-Windows programmers are reinstalled.  C++ programmers become undefined,
   anyone who refers to them will die as well.  Java programmers reincarnate
   after being garbage collected.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
